staking/types/validator.go

```diff
@@ -555,6 +574,16 @@ func UpdateValidatorFromEditMsg(validator *Validator, edit *EditValidator) error
 	return nil
 }
 
+// IsEligibleForEPoSAuction ..
+func IsEligibleForEPoSAuction(validator *ValidatorWrapper) bool {
+	switch validator.EPOSStatus {
+	case effective.FirstTimeCandidate, effective.Candidate:
```

Comment:
For example, I don't see where the validator turn from other status e.g. InCommitteeAndSigning to Candidate? Seems elected validators won't get elected in new epochs. That's the complexity of state transition I am talking about after adding too many different status.

Reply:
It’s true but the status are fleshing out the overloaded way we are using this field , downstream on CLI people were misunderstanding the meaning of active / inactive

Reply:
That first scenario you mention is done in edit validator as default case